src/create-client.test.ts

```diff
@@ -335,6 +346,84 @@ describe("create-client", () => {
           scope.done();
         });
       });
+
+      describe("when the current access token has expired", () => {
+        const clientWithExpiredAccessToken = async () => {
+          const now = Date.now();
+          const { scope: initialRefreshScope } = nockRefresh({
+            accessTokenClaims: {
+              iat: now,
+              // deliberately issuing a JWT that expires at the same time
+              // as iat (we don't trust the client's clock, so back-dating
+              // the timestamps doesn't do anything)
+              exp: now,
+              jti: "initial-access-token",
+            },
+          });
+
+          client = await createClient("client_123abc", {
+            redirectUri: "https://example.com/",
+            // turning off auto refresh so we can ensure that we're
+            // hitting the expired-access token case
+            onBeforeAutoRefresh: () => false,
+          });
+          initialRefreshScope.done();
+
+          return client;
+        };
+
+        it("gets a new access token and returns it", async () => {
+          const client = await clientWithExpiredAccessToken();
+
+          const { scope: refreshScope } = nockRefresh({
+            accessTokenClaims: {
+              jti: "refreshed-token",
+            },
+          });
+          const accessToken = await client.getAccessToken();
+          expect(getClaims(accessToken).jti).toEqual("refreshed-token");
+          refreshScope.done();
+        });
+
+        it("sends a request for each call", async () => {
```

Comment:
this is not actually the behavior we want, but i figured i would test what we have and update the test when i fix it in a subsequent PR.